addons/udes_stock/models/stock_picking.py

```diff
@@ -0,0 +1,223 @@
+        }
+        picking_values.update(kwargs)
+        if not products_info:
+            return [picking_values], products_info
+        else:
+            if any(isinstance(el, list) for el in products_info):
+                num_pickings = len(products_info)
+                picking_vals = [picking_values.copy() for i in range(num_pickings)]
+            else:
+                # Convert picking values to a list of picking values
+                picking_vals = [picking_values]
+                # Convert the products_info to a list of lists
+                products_info = [products_info]
+            return picking_vals, products_info
+
+    def create_picking(
+        self,
+        picking_type,
+        products_info=None,
+        confirm=False,
+        assign=False,
+        create_batch=False,
+        **kwargs,
+    ):
+        """ Create and return a picking for the given picking_type
+            For multiple pickings a list of lists of dicts of product_info should be passed, 
+            and pickings with the same picking_type and other kwargs are the same. 
+            The kwargs are applied to pickings, not moves. If needed, the moves can be created outside of create_pickings with _create_moves
+            
+            
+            :args:
+                - picking_type: picking type of the picking
+            :kwargs:
+                - products_info: list of dicts (or list(list(dicts)) for multiple picks) with product information
+                - confirm: boolean flag to trigger action_confirm
+                - assign: boolean flag to trigger action_assign
+                - create_batch: boolean flag if a batch should be created
+
+        """
+        Picking = self.env['stock.picking']
+
+        # Prepare stock.picking info
+        picking_values, products_info = self._prepare_picking_info(picking_type, products_info=products_info, **kwargs)
+        # Create pickings
+        pickings = Picking.create(picking_values)
+        # Prepare stock.moves
+        if products_info:
+            move_values = self._prepare_move(pickings, products_info)
+            # Create stock.moves
+            self._create_move(move_values)
+            
+        if confirm:
+            pickings.action_confirm()
+
+        if assign:
+            pickings.action_assign()
+
+        if create_batch:
+            self._create_batch(pickings)
+
+        return pickings
+
+    def _create_batch(self, pickings):
+        """ Create batch """
+        PickingBatch = self.env['stock.picking.batch']
+        PickingBatch.create({'picking_ids': [(6, 0, pickings.ids)]})
+
+    def _prepare_move(self, pickings, products_info, **kwargs):
+        """ Return a list of the move details to be used later in creation of the move(s).
+            The purpose of this is to allow for multiple moves to be created at once.
+            Note: pickings and products_info should be formatted via _prepare_picking_info.
```

Comment:
No need for this note, since it is not entirely true. Just describe the format we are expecting for both of them.